### PR TITLE
mon: fix cli commands for mix-version mons

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/mon/MonCommand.h [hunk 1]
```diff
@@ -59,7 +59,7 @@ struct MonCommand {
     encode(helpstring, bl);
     encode(module, bl);
     encode(req_perms, bl);
-    std::string availability;  // Removed field, for backward compat
+    std::string availability = "cli,rest";  // Removed field, for backward compat
```
Comment (gregsfortytwo) [inline src/mon/MonCommand.h]: Is there a mechanism for removing this eventually, or do we just not care?

File: src/common/cmdparse.cc [hunk 2]
```diff
@@ -65,6 +65,55 @@ arg_desc_t cmddesc_get_args(const String& cmddesc)
...
+    // if no , or =, must be a plain word to put out
+    if (word.find_first_of(",=") == string::npos) {
+      out.push_back(word);
+      continue;
+    }
+    auto desckv = cmddesc_get_args(word);
+    auto j = desckv.find("type");
+    if (j != desckv.end() && j->second == "CephBool") {
+      // Instruct legacy clients to send --foo-bar string in place
+      // of a 'true'/'false' value
```
Comment (gregsfortytwo) [inline src/common/cmdparse.cc]: Can you explain what's happening here a bit more? The whole point of the local_mon_commands versus leader_mon_commands is that the peons are supposed to present a unified command view and forward anything they don't understand to the leader. Did we switch the syntax enough on these options that the pre-nautilus monitors just can't pass them through validation?

Comment (liewegas) [reply src/common/cmdparse.cc]: update

**Comment by liewegas** [comment]
@gregsfortytwo ping

**Comment by liewegas** [comment]
retest this please

**Pull request merged**